ambry-replication/src/main/java/com.github.ambry.replication/CloudToStoreReplicationManager.java

```diff
@@ -113,6 +115,8 @@ public CloudToStoreReplicationManager(ReplicationConfig replicationConfig, Clust
         new DiskTokenPersistor(cloudReplicaTokenFileName, mountPathToPartitionInfos, replicationMetrics, clusterMap,
             tokenHelper, storeManager);
     this.localPartitionNameToPartition = mapPartitionNameToPartition(clusterMap, currentNode);
+    this.partitionsToReplicate = replicationConfig.vcrRecoveryPartitions.isEmpty() ? new HashSet<>()
+        : Arrays.stream(replicationConfig.vcrRecoveryPartitions.split(",")).collect(Collectors.toSet());
```

Comment:
If you want to make `Utils.splitString` more flexible, since it seems like you want to put the parts into a set, you can extend the util method to something like this:
```java
  public static ArrayList<String> splitString(String data, String delimiter) {
    return splitString(data, delimiter, ArrayList::new);
  }

  public static <C extends Collection<String>> C splitString(String data, String delimiter,
      Supplier<C> collectionFactory) {
    if (data == null) {
      throw new IllegalArgumentException("Passed in string is null ");
    }
    return Arrays.stream(data.split(delimiter))
        .filter(s -> !s.isEmpty())
        .collect(Collectors.toCollection(collectionFactory));
  }
```

```java
Utils.splitString(vcrRecoveryPartitions, ",", HashSet::new)
```

Reply:
I thought of making this a set and split in the config class itself, but my concern with making anything in config a set is that I will have to write same logic in the closed source code to get the string config and then split it there again. Thats why I moved the split to where it is really used.
I like the suggestion of extending Utils.splitString though.

Reply:
I don't quite understand "writing same logic in closed source code". Let's discuss this offline.